### PR TITLE
Fixups to scheduler/priority settings

### PR DESCRIPTION
**Pull request by kolyshkin**
These commits amend commit bfbd0305 (#3783) and 770728e1 (#4025). Should not result in any visible change, except:
 - invalid `IOPriority.Class` value will result in an error much earlier;
 - `runc exec` changes IO priority later, just before exec;
 - `configs.IOPrioClassMapping` map is removed from the public API.
 
 Should be relatively easy to review.

### CODE REVIEW

**Review by kolyshkin — COMMENTED** [review]
@opencontainers/runc-maintainers @AkihiroSuda @cyphar @thaJeztah @rata @lifubang can we please review/merge this? This will unblock my further work (adding CPU affinity).

File: libcontainer/configs/validate/validator.go
Comment (lifubang) [inline libcontainer/configs/validate/validator.go:412]: I don't like this comment, but this is my personal preference. Maybe should change to `// should return nil` or move to an new function:
```go
func checkIOPriorityClass(specs.IOPriorityClass) bool {
    switch config.IOPriority.Class {
	case specs.IOPRIO_CLASS_RT, specs.IOPRIO_CLASS_BE, specs.IOPRIO_CLASS_IDLE:
            return true
    }
    return false
}
```

**Review by lifubang — APPROVED** [review]
LGTM
Left two unimportant comments.

**Review by AkihiroSuda — APPROVED** [review]

File: libcontainer/configs/validate/validator.go
Comment (kolyshkin) [reply libcontainer/configs/validate/validator.go:412]: Changed the comment to ` // Valid class, do nothing.`, and the whole switch to
```go
          switch class := config.IOPriority.Class; class {                    
          case specs.IOPRIO_CLASS_RT, specs.IOPRIO_CLASS_BE, specs.IOPRIO_CLASS_IDLE:
                  // Valid class, do nothing.
          default:
                  return fmt.Errorf("invalid ioPriority.Class: %q", class)                                             
          }
```

File: libcontainer/setns_init_linux.go
Comment (kolyshkin) [reply libcontainer/setns_init_linux.go:78]: Makes sense (although in case of `setupScheduler` we pass the whole config because we also need to access config.Cgroups).

I've also changed its name, as other similar functions have `setup` not `set` prefix.

The change:
```diff
 
-func setIOPriority(ioprio *configs.IOPriority) error {
+func setupIOPriority(config *configs.Config) error {
        const ioprioWhoPgrp = 1
 
+       ioprio := config.IOPriority
        if ioprio == nil {
                return nil
        }
```

**Review by lifubang — APPROVED** [review]

**Pull request merged**
